GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,226 @@
+            if (string.IsNullOrEmpty(this.Directory) || this.Directory.Equals("."))
+            {
+                if (Environment.CurrentDirectory.StartsWith(enlistment.WorkingDirectoryRoot))
+                {
+                    this.Directory = Environment.CurrentDirectory.Substring(enlistment.WorkingDirectoryRoot.Length);
+                }
+                else
+                {
+                    // If the path is not under the source root, set the directory to empty
+                    this.Directory = string.Empty;
+                }
+            }
+
+            this.Directory = this.Directory.Replace('\\', GVFSConstants.GitPathSeparator);
+
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData pathData = new GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData();
+
+            this.GetPathsFromGitIndex(enlistment, pathData);
+            this.GetPlaceholdersFromDatabase(enlistment, pathData);
+            this.GetModifiedPathsFromPipe(enlistment, pathData);
+
+            pathData.NormalizeAllPaths();
+
+            GVFSEnlistmentHealthCalculator enlistmentHealthCalculator = new GVFSEnlistmentHealthCalculator(pathData);
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData = enlistmentHealthCalculator.CalculateStatistics(this.Directory);
+
+            this.PrintOutput(enlistmentHealthData);
+        }
+
+        private void PrintOutput(GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData)
+        {
+            string trackedFilesCountFormatted = enlistmentHealthData.GitTrackedItemsCount.ToString("N0");
+            string placeholderCountFormatted = enlistmentHealthData.PlaceholderCount.ToString("N0");
+            string modifiedPathsCountFormatted = enlistmentHealthData.ModifiedPathsCount.ToString("N0");
+
+            // Calculate spacing for the numbers of total files
+            int longest = Math.Max(trackedFilesCountFormatted.Length, placeholderCountFormatted.Length);
+            longest = Math.Max(longest, modifiedPathsCountFormatted.Length);
+
+            // Sort the dictionary to find the most hydrated directories by percentage
+            List<KeyValuePair<string, decimal>> topLevelDirectoriesByHydration = enlistmentHealthData.DirectoryHydrationLevels.Take(this.DirectoryDisplayCount).ToList();
+
+            this.Output.WriteLine("\nHealth of directory: " + enlistmentHealthData.TargetDirectory);
+            this.Output.WriteLine("Total files in HEAD commit:           " + trackedFilesCountFormatted.PadLeft(longest) + " | 100%");
+            this.Output.WriteLine("Files managed by VFS for Git (fast):  " + placeholderCountFormatted.PadLeft(longest) + " | " + this.FormatPercent(enlistmentHealthData.PlaceholderPercentage));
+            this.Output.WriteLine("Files managed by git (slow):          " + modifiedPathsCountFormatted.PadLeft(longest) + " | " + this.FormatPercent(enlistmentHealthData.ModifiedPathsPercentage));

```

Comment:
What does `fast` vs `slow` mean?

Reply:
I was having trouble coming up with better wording. The `fast` files (placeholders) are slower than files not on disk, but faster than the `slow` files (modified paths and skip-worktree off) which have the worst performance